SilentUpdate/src/main/java/org/netbeans/modules/autoupdate/silentupdate/UpdateHandler.java

```diff
@@ -319,10 +350,115 @@ static Installer doVerify(InstallSupport support, Validator validator) throws Op
 
     static Restarter doInstall(InstallSupport support, Installer installer) throws OperationException
     {
+        OperationContainer<OperationSupport> con = OperationContainer.createForDirectUninstall();
         return support.doInstall(installer, null);
     }
 
-    private static boolean installNewModules()
+    static void doDisable(Collection codeNames)
+    { // codeName contains code name of modules for disable
+        Collection<UpdateElement> toDisable = new HashSet();
+        List<UpdateUnit> allUpdateUnits = UpdateManager.getDefault().getUpdateUnits(UpdateManager.TYPE.MODULE);
+        for (UpdateUnit unit : allUpdateUnits)
+        {
+            if (unit.getInstalled() != null)
+            { // filter all installed modules
+                UpdateElement el = unit.getInstalled();
+                if (el.isEnabled())
+                { // filter all enabled modules
+                    if (codeNames.contains(el.getCodeName()))
+                    { // filter given module in the parameter
+                        toDisable.add(el);
+                    }
+                }
+            }
+        }
+
+        OperationContainer<OperationSupport> oc = OperationContainer.createForDirectDisable();
+
+        for (UpdateElement module : toDisable)
+        {
+            if (oc.canBeAdded(module.getUpdateUnit(), module))
+            { // check if module can be disabled
+                OperationInfo operationInfo = oc.add(module);
+                if (operationInfo == null)
+                { // it means it's already planned to disable
+                    continue;
+                }
+                // get all module depending on this module
+                Set requiredElements = operationInfo.getRequiredElements();
+                // add all of them between modules for disable
+                oc.add(requiredElements);
+            }
+        }
+
+        // check the container doesn't contain any invalid element
+        assert oc.listInvalid().isEmpty();
```

Comment:
we halt execution because ``if (oc.canBeAdded(module.getUpdateUnit(), module))`` suggests that there should never be any invalid elements at this point?